scylla/src/transport/execution_profile.rs

```diff
@@ -0,0 +1,469 @@
+}
+
+impl ExecutionProfile {
+    pub(crate) fn new_from_inner(inner: ExecutionProfileInner) -> Self {
+        Self(Arc::new(inner))
+    }
+
+    /// Creates a blank builder that can be used to construct new ExecutionProfile.
+    pub fn builder() -> ExecutionProfileBuilder {
+        ExecutionProfileBuilder {
+            request_timeout: None,
+            consistency: None,
+            serial_consistency: None,
+            load_balancing_policy: None,
+            retry_policy: None,
+            speculative_execution_policy: None,
+        }
+    }
+
+    /// Creates a builder having all options set to the same as set in this ExecutionProfile.
+    pub fn to_builder(&self) -> ExecutionProfileBuilder {
+        ExecutionProfileBuilder {
+            request_timeout: Some(self.0.request_timeout),
+            consistency: Some(self.0.consistency),
+            serial_consistency: Some(self.0.serial_consistency),
+            load_balancing_policy: Some(self.0.load_balancing_policy.clone()),
+            retry_policy: Some(self.0.retry_policy.clone()),
+            speculative_execution_policy: Some(self.0.speculative_execution_policy.clone()),
+        }
+    }
+
+    /// Returns a new handle to this ExecutionProfile.
+    pub fn into_handle(self) -> ExecutionProfileHandle {
+        ExecutionProfileHandle(Arc::new((ArcSwap::new(self.0), None)))
+    }
+
+    /// Returns a new handle to this ExecutionProfile, tagging the handle with provided label.
+    /// The tag, as its name suggests, is only useful for debugging purposes, while being confused
+    /// about which statement/session is assigned which handle. Identifying handles with tags
+    /// could then help.
+    pub fn into_handle_with_label(self, label: String) -> ExecutionProfileHandle {
+        ExecutionProfileHandle(Arc::new((ArcSwap::new(self.0), Some(label))))
+    }
+}
+
+/// A handle that points to an ExecutionProfile.
+///
+/// Its goal is to enable remapping all associated entities (query/Session)
+/// to another execution profile at once.
+/// Note: Cloned handles initially point to the same Arc'ed execution profile.
+/// However, as the mapping has yet another level of indirection - through
+/// `Arc<ArcSwap>` - remapping one of them affects all the others, as under the hood
+/// it is done by replacing the Arc held by the ArcSwap, which is shared
+/// by all cloned handles.
+/// The optional String is just for debug purposes. Its purpose is described
+/// in [ExecutionProfile::into_handle_with_label].
+#[derive(Debug, Clone)]
+pub struct ExecutionProfileHandle(Arc<(ArcSwap<ExecutionProfileInner>, Option<String>)>);
+
+impl ExecutionProfileHandle {
+    pub(crate) fn access(&self) -> Arc<ExecutionProfileInner> {
+        self.0 .0.load_full()
+    }
+
+    /// Makes the handle point to a new execution profile.
+    /// All entities (queries/Session) holding this handle will reflect the change.
+    pub fn map_to_another_profile(&mut self, profile: &ExecutionProfile) {
```

Comment:
```suggestion
    pub fn map_to_another_profile(&mut self, profile: ExecutionProfile) {
```

Would it make sense for `map_to_another_profile` to consume `ExecutionProfile`? It's a similar situation to `set_execution_profile_handle` which consumes a handle.

Reply:
I do not oppose this, makes sense.

Reply:
Makes sense to me as well.